flock_agent/daemon/daemon.py

```diff
@@ -135,16 +146,31 @@ def log(self, request, response, time):
         async def set_setting(request):
             key = request.match_info.get("key", None)
             val = await request.json()
-            self.global_settings.set(key, val)
-            self.global_settings.save()
-            return response_object()
 
-        async def get_twig(request):
-            twig_id = request.match_info.get("twig_id", None)
-            try:
-                return response_object(self.global_settings.get_twig(twig_id))
-            except:
-                return response_object(error="invalid twig_id")
+            # Only change the setting if it's actually changing
+            old_val = self.global_settings.get(key)
+            if old_val == val:
+                self.c.log(
+                    "Daemon",
+                    "http_server.set_settings",
+                    "skipping {}={}, because it's already set".format(key, val),
+                )
+            else:
+                self.c.log(
+                    "Daemon",
+                    "http_server.set_settings",
+                    "setting {}={}".format(key, val),
+                )
+                self.global_settings.set(key, val)
+                self.global_settings.save()
+
+                if key == "use_server":
```

Comment:
If you don't have to disable and re-enable the server to change the sever endpoint, we should also add that one.

Reply:
At the moment the only way to change the server endpoint is to actually delete the settings file, and it's undocumented how. I think it would make sense to make it possible for users to change the server but that can be for a separate issue